include/rocksdb/file_system.h

```diff
@@ -825,6 +852,20 @@ class FSRandomAccessFile {
     return IOStatus::NotSupported("InvalidateCache not supported.");
   }
 
+  //  Asynchronous Read of the data requested in FSReadRequest. This is a
+  //  asynchronous call, i.e it will return after submitting the request.
+  //  When read request is completed callback function specified in cb will be
+  //  called with result populated in FSReadResponse. This API also populates
+  //  IOHandle. req contains the request offset and size, but the result and
+  //  status fields should be ignored. Instead, they should be passed in the
+  //  FSReadResponse structure to the callback.
+  virtual IOStatus ReadAsync(const IOOptions& /*opts*/, IODebugContext* /*dbg*/,
+                             FSReadRequest* /*req*/,
+                             std::function<void(FSReadResponse* resp)> /*cb*/,
```

Comment:
ReadAsync should also take a callback argument (void*) that can be passed back to the cb. It'll be useful for identifying the request associated with the response.

Reply:
This comment is not resolved yet. ReadAsync should look something like this -

```
virtual IOStatus ReadAsync(const IOOptions& opts, const FSReadRequest& req, std::function<void(const FSReadResponse& resp, void* cb_arg)> cb, void* cb_arg)
```

The cb_arg will allow RocksDB to correlate the callback with the original request.